torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -890,10 +889,14 @@ class CheckpointHigherOrderVariable(WrapHigherOrderVariable):
     def call_function(
         self, tx, args: List[VariableTracker], kwargs: Dict[str, VariableTracker]
     ) -> VariableTracker:
-        from torch._higher_order_ops.wrap import TagActivationCheckpoint
+        from torch._higher_order_ops.wrap import HandleActivationCheckpoint
         from .builder import wrap_fx_proxy
 
-        checkpoint_kwargs, gmod_kwargs = TagActivationCheckpoint.divide_kwargs(kwargs)
+        if "context_fn" in kwargs:
```

Comment:
So this works with arbitrary context_fns?

Reply:
Only guarantees to work with TorchDispatch based context_fns actually. Maybe we can assert that user must use TorchDispatch functions?

Reply:
Added check in torch/utils/checkpoint.py and corresponding unit test.